### ISSUE TITLE
Scrollbar visible without content

### ISSUE THREAD

**Issue by vishalsingh119** [zefoy/ngx-perfect-scrollbar#293]
I am using a perfect-scrollbar in angular material modal. as you can see in the screenshot, right side doesn't have any content after that perfect scrollbar coming. It disappears when I scroll down in that part.

Screenshot- https://ibb.co/VxvSxjy

**Comment by Art2058** [comment]
Same issue

**Comment by sconix** [comment]
You might need to call the perfect-scrollbar update method after dynamic Angular changes that perfect-scrollbar does not pick.

**Comment by vishalsingh119** [comment]
it's not about adding the perfect scrollbar after the content update. the issue is without any content scrollbar appear

**Comment by sconix** [comment]
Yes but the reason for is the perfect scrollbar being initialized before DOM is ready. Perfect scrollbar is a pure JS library so it does not know anything about shadow DOM etc. techniques Angular uses, so it needs to be told when DOM has been updated so that it can adjust itself i.e. hide the scrollbars if there is no content and so on. I recommend you to look into ngx-scrollbar library instead which is an Angular library and does better job in many things than this library.

**Comment by vishalsingh119** [comment]
@sconix I am using perfect scrollbar at many places in the same page, so controlling through toggle variable is a messy thing and the ngx-scrollbar library does not support IE11 Browser.

**Comment by jacknkandy** [comment]
I'm also having the same issue. Seems to be after upgrading to v10.

When I hover over a container the scrollbars will be visible until i attempt to scroll or resize the page. Then they will disappear and wont return.

It appears that in these instances the `ps--active-x` and `ps--active-y` classes are applied when the component is rendered.

I have tried waiting for component render and calling the `update()` on the perfect scrollbar element but it didn't seem doesn't fix the issue.

**Comment by Dolgovec** [comment]
The problem does exist. If you are still looking for solution, I haven't found a normal one. So did a hack like this:

HTML:
```
    <div class="ps init-hidden" [perfectScrollbar] #scroll>
        ....
    </div>
```

TS:
```
   ...
  @ViewChild(PerfectScrollbarDirective, {static: false}) scrollArea?: PerfectScrollbarDirective;
  @ViewChild('scroll') scrollRef: ElementRef;
  
  ngOnInit(): void {
      
    ...
    this.removeScroll();
  }

 // hack to remove scroll after modal inits
  removeScroll(): void {
    setTimeout(() => {
      this.scrollArea.update();
      this.scrollRef.nativeElement.classList.remove('init-hidden');
    }, 300);
  }

```

CSS:
```
.ps.init-hidden {
  .ps__rail-x, .ps__rail-y {
    opacity: 0 ;
    transition: all .3s ease;
  }
}
```

I know that it's a bad solution, but at least it works. If you have any better idea how to solve this - share please.

**Comment by Halynsky** [comment]
This works for me
```
  @ViewChild('perfectScroll') perfectScroll: PerfectScrollbarComponent;

  ngAfterViewInit(): void {
    setTimeout(() => {
      this.perfectScroll.directiveRef.update()
    })
  }
```

**Comment by Halynsky** [comment]
```
<perfect-scrollbar [config]="{suppressScrollX : true, suppressScrollY : true}" #perfectScroll>
</perfect-scrollbar>
```

```
@ViewChild('perfectScroll') perfectScroll: PerfectScrollbarComponent;

  ngAfterViewInit(): void {
    setTimeout(() => {
      this.perfectScroll.directiveRef.update()
      this.perfectScroll.directiveRef.config.suppressScrollY = false;
    }, 300)
  }
```
